rd-net/RdFramework/Impl/SocketWire.cs

```diff
@@ -617,7 +598,7 @@ public ServerFactory(Lifetime lifetime, IScheduler scheduler, IPEndPoint endpoin
         IPEndPoint endpoint = null
       )
       {
-        var serverSocket = Server.CreateServerSocket(lifetime, endpoint);
+        var serverSocket = Server.CreateServerSocket(endpoint);
```

Comment:
ServerFactory's serverSocket lifetime should be greater than lifetimes of individual Server instances it creates. As this is written right now, ServerFactory's socket will be closed after the first Server disconnects, which is not the desired behavior.

Reply:
It does not work as you described (lifetime is common), but I have pushed theoretical fix.